404.html

```diff
@@ -18,6 +18,42 @@
   </script>
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <meta property="og:title" content="Page not found">
+  <script type="module">
```

Comment:
You can skip this part and hardcode the paths. For multi store we likely would need to have localized 404 pages which then get assigned via CDN. Something like `/en/404` and `/fr/404` documents which get mapped to the 404 status code using Fastly VCL.

Reply:
I'm trying to avoid the need for code changes to add a new language and have that all be driven by the CMS, hence why the `root` is set at the metadata level. Aside from the folder mapping used for PDP, it's almost possible, but I digress.

What's the harm in having this dynamically set here other than optimizing the 404 page? Perhaps we can add a condition so the fetch is skipped completely if the metadata is present, i.e., already optimized at the CDN level.

Reply:
For multi language the 404s would need to be driven by the CMS yes. 

> What's the harm in having this dynamically set here other than optimizing the 404 page?

No harm, but that 404 wouldn't be localized. Edge Delivery delivers the same `404.html` page regardless of the current path which makes it unfeasible for production use cases.